src/dibit_vector.jl

```diff
@@ -0,0 +1,70 @@
+"""
+    DiBitVector <: AbstractVector{UInt8}
+
+A bitvector whose elements are two bits wide, allowing
+storage of integer values between 0 and 3. Optimized
+for performance and memory savings.
+"""
+mutable struct DiBitVector <: AbstractVector{UInt8}
```

Comment:
What about `<: AbstractVector{Int8}` ? (not that I prefer `Int8`, but was just curious why you chose `UInt8`).

Reply:
I chose `UInt8` because of my use case: a coloring from 0 to 3.  I suppose we could change it to `Int8` for the folks who want -2 to +1... but I don't know of any use cases for that offhand.

Reply:
Ok, but I was more thinking `0` to `3` (it would just be a matter of adding `% Int8` in `getindex`.

Reply:
My thought was because it is a subtype of `AbstractVector{UInt8}` it should output `UInt8` as the eltype.

I made it a subtype of `UInt8` because, essentially, that's what it is. (well, 8 of them packed into a `UInt64`, but still - unsigned.)

Reply:
Sure if it's a subtype of `AbstractVector{UInt8}`, it must return `UInt8`. But this would work equally well with `Int8`. I guess it depends on the use-case. Setting that as a type parameter would be an option (`mutable struct DiBitVector{T<:Integer} <: AbstractVector{T}`), but might be overkill, I don't know.

Reply:
I think it's overkill. `BitVector` is `<: AbstractArray{Bool, 1}`, which makes sense. The smallest valid type for a `DiBitVector` is `UInt8`. (If we had nybbles, I'd use that.) `Int8` is the same width but is ambiguous: do we range from -2 to +1, or do we range from 0 to 3?